spring-rabbit/src/test/java/org/springframework/amqp/rabbit/listener/SimpleMessageListenerContainerTests.java

```diff
@@ -736,26 +736,28 @@ void testShutdownWithPendingReplies() {
 
 		template.setReplyAddress(container.getQueueNames()[0]);
 
-		long shutdownTimeout = 2000L;
+		long shutdownTimeout = 500L;
 		container.setShutdownTimeout(shutdownTimeout);
 
-		ActiveObjectCounter<Object> replyCounter =
-				(ActiveObjectCounter<Object>) ReflectionTestUtils.getField(template, "pendingRepliesCounter");
+		ActiveObjectCounter<Object> replyCounter = template.getPendingReplyCounter();
 		assertThat(replyCounter).isNotNull();
 
 		Object pending = new Object();
 		replyCounter.add(pending);
 		assertThat(replyCounter.getCount()).isEqualTo(1);
 
+		Log logger = spy(TestUtils.getPropertyValue(container, "logger", Log.class));
+		new DirectFieldAccessor(container).setPropertyValue("logger", logger);
+
 		container.start();
 
-		long startTime = System.currentTimeMillis();
 		container.stop();
-		long stopDuration = System.currentTimeMillis() - startTime;
 
-		replyCounter.release(pending);
+		await().atMost(Duration.ofMillis(500)).untilAsserted(() ->
```

Comment:
I think those max 30 seconds by default is fine with us in our tests.
It is really not going to be blocked that long.
Might really maximum those 500 millis.
However, if we have it as exact our given time, there is no guarantee that CPU resources will be available for us in time to satisfy this expectation.